### DISCUSSION TITLE
44. 라우터, 스위치, 허브 장치에 대해 설명해주세요.

### DISCUSSION THREAD

**Discussion by linirini** [jewelry-box/cs-interview#45] [General]
.

**Comment by eun-byeol** [comment]
`라우터`는 주로 `네트워크 계층(3계층)`에서 사용되고, `네트워크(LAN, WAN) 간 연결`하고 `IP 주소` 기반으로  `패킷의 최적 경로`를 선택하여 전달합니다.

`스위치`는 주로 `데이터 링크 계층(2계층)`에서 사용되고, `같은 네트워크(LAN)`에서 `MAC 주소` 기반으로 프레임을 전달합니다. (L3 스위치의 경우 네트워크 계층에서도 사용됩니다.)

`허브`는 `물리 계층(1계층)`에서 사용되고, `같은 네트워크(LAN)`에 연결된 모든 장치로 `브로드캐스트` 방식으로 데이터를 전송합니다.

---
> 스위치

- MAC 주소 테이블 포함
- 테이블에 없을 경우 모든 장치에 데이터 전송(Flooding)
![image](https://github.com/user-attachments/assets/6293da5a-2938-4e7f-853c-3dcc67da2a18)

> 허브

- LAN 구성 시 가까운 장비들을 케이블을 사용하여 연결
- 리피터(신호 증폭)와 멀티포트(중심에서 여러 컴퓨터 연결) 역할
- 브로드캐스트 방식으로 전송하기 때문에 과도한 트래픽 발생한다는 단점 존재

![image](https://github.com/user-attachments/assets/a5543eab-b019-43cd-8b53-7a62ad74a3b8)
